src/integrations/windowsIntegrationManager.ts

```diff
@@ -358,60 +388,65 @@ export default class WindowsIntegrationManager implements IntegrationManager {
     console.log(`kubeconfig integration for ${ distro } set to ${ state }`);
   }
 
-  /**
-   * List the registered WSL2 distributions.
-   */
-  protected get distros(): Promise<string[]> {
+  protected get externalDistros(): Promise<WSLDistro[]> {
     return (async() => {
-      const distros = (await this.captureCommand(
-        { encoding: 'utf16le' },
-        '--list', '--quiet'))
-        .split(/[\r\n]+/g)
-        .map(x => x.trim())
-        .filter(x => x);
-
-      if (distros.length < 1) {
-      // Return early if we find no distributions in this list; listing again
-      // with verbose will fail if there are no distributions.
-        return [];
-      }
-
-      const stdout = await this.captureCommand({ encoding: 'utf16le' }, '--list', '--verbose');
+      const wslOutput = await this.captureCommand({ encoding: 'utf16le' }, '--list', '--verbose');
       // As wsl.exe may be localized, don't check state here.
       const parser = /^[\s*]+(?<name>.*?)\s+\w+\s+(?<version>\d+)\s*$/;
-      const result = stdout.trim()
+      const distros = wslOutput.trim()
         .split(/[\r\n]+/)
         .slice(1) // drop the title row
-        .map(line => line.match(parser))
+        .map(line => line.match(parser)?.groups)
         .filter(defined)
-        .filter(result => result.groups?.version === '2')
-        .map(result => result.groups?.name)
-        .filter(defined);
+        .map(group => new WSLDistro(group.name, parseInt(group.version)))
+        .filter((distro: WSLDistro) => !DISTRO_BLACKLIST.includes(distro.name));
 
-      return result.filter(x => distros.includes(x)).filter(x => !DISTRO_BLACKLIST.includes(x));
+      return distros;
     })();
   }
 
+  protected get validExternalDistros(): Promise<WSLDistro[]> {
```

Comment:
I guess that's "external distros, but only those that we can possibly integrate with"? How do you feel about calling it "supported" instead of "valid"?

Reply:
I think "supported" is better... I'm going to add a brief comment, too.